lib/sles4sap/sap_deployment_automation_framework/basetest.pm

```diff
@@ -95,6 +96,32 @@ sub full_cleanup {
 
 sub post_fail_hook {
     record_info('Post fail', 'Executing post fail hook');
+    if (testapi::is_serial_terminal()) {
+        if ($serial_regexp_playbook) {
+            # Teminate the process when playbook timed out
+            # Note: please do not use '> ' (or $testapi::distri->{serial_term_prompt})
+            # as there are lots of '> ' in the oupput of playbook
+            unless (wait_serial(qr/-\d+-/)) {
+                type_string('', terminate_with => 'ETX');
+                # Wait 5s for process returns and serial terminal gets ready
+                sleep 5;
```

Comment:
Wouldn't something like 
```
wait_serial($testapi::distri->{serial_term_prompt}
# or
wait_serial(qr/:~|#|>/, quiet => 1);
```
work here better? It would detect command line directly.

Reply:
I would like to keep the `sleep 5` as the outputs of playbook contains `> `, using `wait_serial` is a risk. 
Adding `sleep 5` just for the show of serial0.txt (for example) is a little bit readable as the playbook outputs messed up with `type_string()` here.
Add, we can also delete `sleep 5` if you don't like it as it is useless for fixing the post fail hook issue..

Reply:
JFYI: 
After adding 'sleep 5' the serial0.txt looks like:
`^Chj2Hk-120-Comment: Executing playbook: playbook_04_00_00_db_install.yaml`
Before adding `sleep `  the serial0.txt looks like:
```
^C

whoami | grep root; echo 9Zjcg-$?-

m8ntV-120-Comment: Executing playbook: playbook_04_00_00_db_install.yaml
```

Reply:
I will try to delete `sleep` and using  `wait_serial(qr/-\d+-/)` (`qr/-\d+-/` is safe).

Reply:
> I will try to delete `sleep` and using `wait_serial(qr/-\d+-/)` (`qr/-\d+-/` is safe).

Updated code (replaced `sleep 5` with `wait_serial(qr/-\d+-/);`) and the VR is good: https://openqaworker15.qa.suse.cz/tests/302824#step/deploy_hanasr/458